### PR TITLE
RHOAIENG-1124: Initial pipelines list page test

### PR DESCRIPTION
**Pull request by jpuzz0**
## Description
Initial test for the PipelinesList page which simply loads the page and renders the PipelinesTable with mock data.

## How Has This Been Tested?
Running `cypress:run` or `cypress:open` and selecting `PipelinesList.cy.ts`

### CODE REVIEW

File: frontend/src/__tests__/cypress/cypress/pages/components/PipelinesTable.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+export class PipelinesTable {
+  private testId = 'pipelines-table';
+
+  find(): Cypress.Chainable<JQuery<HTMLElement>> {
+    return cy.findByTestId(this.testId);
+  }
+}
+
+export const pipelinesTable = new PipelinesTable();
```
Comment (christianvogt) [inline frontend/src/__tests__/cypress/cypress/pages/components/PipelinesTable.ts:9]: `pages/components` is meant to be a folder of generic UI utilities.
We already have a `pages/pipelines.ts` file for pipeline related page objects.

File: frontend/src/__tests__/cypress/cypress/e2e/pipelines/PipelinesList.cy.ts [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+
+const projectName = 'test-project-name';
+
+describe('PipelinesList', () => {
+  beforeEach(() => {
+    initIntercepts();
+  });
+
+  it('renders the page with pipelines table data', () => {
+    cy.visitWithLogin(`/pipelines/${projectName}`);
```
Comment (christianvogt) [inline frontend/src/__tests__/cypress/cypress/e2e/pipelines/PipelinesList.cy.ts:26]: Make use of a page object for visiting the pipelines page.

File: frontend/src/__tests__/cypress/cypress/pages/pipelinesList.ts [hunk 3]
```diff
@@ -0,0 +1,19 @@
+class PipelinesList {
+  private testId = 'pipelines-list-page';
+
+  visit(projectName: string) {
+    cy.visitWithLogin(`/pipelines/${projectName}`);
+    this.wait();
+  }
+
+  wait() {
```
Comment (christianvogt) [inline frontend/src/__tests__/cypress/cypress/pages/pipelinesList.ts:9]: Pattern is for `wait` to be `private`.

File: frontend/src/pages/projects/screens/detail/pipelines/PipelinesSection.tsx [hunk 4]
```diff
@@ -38,6 +38,7 @@ const PipelinesSection: React.FC = () => {
         isEmpty={!installed}
         emptyState={<NoPipelineServer variant="secondary" />}
         showDivider={isPipelinesEmpty}
+        data-testid="pipelines-list-page"
```
Comment (christianvogt) [inline frontend/src/pages/projects/screens/detail/pipelines/PipelinesSection.tsx:41]: The `PipelinesSection` is not used as a standalone page.
This test id should be more appropriately named `pipelines-section`.

File: frontend/src/__tests__/cypress/cypress/pages/pipelinesList.ts [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+    this.find();
+    cy.testA11y();
+  }
+
+  find() {
+    return cy.findByTestId(this.testId);
+  }
```
Comment (christianvogt) [inline frontend/src/__tests__/cypress/cypress/pages/pipelinesList.ts:16]: `find` function name should be have a suffix describing what is being found. What are we finding? A general `find` is only useful when establishing a context where all subsequent selectors are chained from this. See `pages/component/Modal.ts` for example.

Is the `PipelinesSection` even used outside of project details? All selectors in a page object must be relevant to the page referenced by the `visit` function.

File: frontend/src/__tests__/cypress/cypress/e2e/pipelines/PipelinesList.cy.ts [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+      pathname: `/api/k8s/apis/datasciencepipelinesapplications.opendatahub.io/v1alpha1/namespaces/${projectName}/datasciencepipelinesapplications/pipelines-definition`,
+    },
+    mockDataSciencePipelineApplicationK8sResource({ namespace: projectName }),
+  );
+  cy.intercept(
+    {
+      pathname: `/api/k8s/apis/kubeflow.org/v1/namespaces/${projectName}/notebooks`,
+    },
+    mockK8sResourceList([mockNotebookK8sResource({ namespace: projectName })]),
+  );
```
Comment (christianvogt) [inline frontend/src/__tests__/cypress/cypress/e2e/pipelines/PipelinesList.cy.ts:45]: Are notebooks needed to be mocked here?

**Review by christianvogt — COMMENTED** [review]
You should be running `cypress:run:mock` or `cypress:open:mock`.
We use the `mock` variant for running tests in CI.

File: frontend/src/__tests__/cypress/cypress/pages/components/PipelinesTable.ts (hunk 1 above)
Comment (jpuzz0) [reply frontend/src/__tests__/cypress/cypress/pages/components/PipelinesTable.ts:9]: Ok, I didn't realize we wanted to put multiple page objects in single files. I'll do this, but will that not make things a bit more difficult to navigate?

File: frontend/src/__tests__/cypress/cypress/e2e/pipelines/PipelinesList.cy.ts (hunk 6 above)
Comment (jpuzz0) [reply frontend/src/__tests__/cypress/cypress/e2e/pipelines/PipelinesList.cy.ts:45]: I don't know the ins and outs of the KF API calls, but without this, the `secrets` call on the page fails for the project name selected.

File: frontend/src/__tests__/cypress/cypress/pages/components/PipelinesTable.ts (hunk 1 above)
Comment (christianvogt) [reply frontend/src/__tests__/cypress/cypress/pages/components/PipelinesTable.ts]: Feel free to have separate files for different page objects. The main point here is that `components` was for generic page objects. Whereas specific ones should be in `pages/` or some other sub dir (which do not exist today).

File: frontend/src/__tests__/cypress/cypress/pages/pipelinesList.ts (hunk 5 above)
Comment (christianvogt) [reply frontend/src/__tests__/cypress/cypress/pages/pipelinesList.ts]: At the time of writing my comment, the related `testId` referred to the `PipelinesSection` which is why I made that comment. If it's meant to target a list, likely the test ID was put in the wrong spot.

Comment (jpuzz0) [reply frontend/src/__tests__/cypress/cypress/pages/pipelinesList.ts]: Got it. I hoisted the test ID so I could get access to the project dropdown when navigating to the pipelines global page (I've been calling it the "last page" - but just the page we show when navigating to "Pipelines" from the left nav). As a result, I updated the page object name and test ID. I think the setup now makes sense, but please let me know if something is still off.

File: frontend/src/__tests__/cypress/cypress/e2e/pipelines/PipelinesList.cy.ts (hunk 6 above)
Comment (jpuzz0) [reply frontend/src/__tests__/cypress/cypress/e2e/pipelines/PipelinesList.cy.ts]: Yeah, sometimes folks want to see all things indicating success even with mocks, but I'm onboard with only mocking if necessary for the test(s). 

Also, once you sent that comment last week I started using that "mock" command instead locally, so the most recent changes were tested/written while using that command instead.

File: frontend/src/__tests__/cypress/cypress/pages/components/PipelinesTable.ts (hunk 1 above)
Comment (jpuzz0) [reply frontend/src/__tests__/cypress/cypress/pages/components/PipelinesTable.ts]: Ok, I'll keep that in mind when individual files seem to be getting too lengthy, I think creating sub-directories to organize pages and their supporting page objects is the way to go.

**Review by DaoDaoNoCode — APPROVED** [review]
/lgtm

**Pull request merged**